core/src/main/java/com/digitalpebble/stormcrawler/protocol/file/FileResponse.java

```diff
@@ -0,0 +1,178 @@
+	static final org.slf4j.Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+	public FileResponse(String u, Metadata md, FileProtocol fp)
+			throws MalformedURLException, IOException {
+		
+		fileProtocol = fp;
+		metadata = md;
+
+		URL url = new URL(u);
+
+		if (LOG.isTraceEnabled()) {
+			LOG.trace("Fetching {}.",url.getPath());
+		}
+
+		if (url.getPath() != url.getFile() && LOG.isWarnEnabled()) {
+			LOG.warn("url.getPath() != url.getFile(): {}.",url);
+		}
+
+		String path = "".equals(url.getPath()) ? "/" : url.getPath();
+
+		File file=new File(URLDecoder.decode(path, fileProtocol.getEncoding()));
+		
+		if (!file.exists()) {
+			statusCode = HttpStatus.SC_NOT_FOUND;
+		}
+
+		if (!file.canRead()) {
+			statusCode = HttpStatus.SC_UNAUTHORIZED;
+		}
+
+		if (!file.equals(file.getCanonicalFile())) {
+			metadata.setValue(HttpHeaders.LOCATION, file.getCanonicalFile().toURI().toURL().toString());
+			statusCode = HttpStatus.SC_MULTIPLE_CHOICES;
+		}
+
+		if (file.isDirectory()) {
+			getDirAsHttpResponse(file);
+		} else if (file.isFile()) {
+			getFileAsHttpResponse(file);
+		} else {
+			statusCode = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+		}
+	}
+
+	public ProtocolResponse toProtocolResponse() throws IOException {
+
+		return new ProtocolResponse(content, statusCode, metadata);
+	}
+
+	@SuppressWarnings("static-access")
+	private void getFileAsHttpResponse(File file) throws IOException {
+		long size = file.length();
+
+		if (size > Integer.MAX_VALUE) {
+			statusCode = HttpStatus.SC_BAD_REQUEST;
+			return;
+		}
+
+		int len = (int) size;
+
+		if (fileProtocol.getMaxContentLength() >= 0 && len > fileProtocol.getMaxContentLength())
+			len = fileProtocol.getMaxContentLength();
+
+		content = new byte[len];
+
+		InputStream is = new FileInputStream(file);
+		int offset = 0;
+		int n = 0;
+		while (offset < len && (n = is.read(content, offset, len - offset)) >= 0) {
+			offset += n;
+		}
+
+		if (offset < len && LOG.isWarnEnabled()) {
+			LOG.warn("not enough bytes read from file: {}",file.getPath());
+		}
+		is.close();
+
+		metadata.setValue(HttpHeaders.CONTENT_LENGTH, new Long(size).toString());
+		metadata.setValue(HttpHeaders.LAST_MODIFIED, formatDate(file.lastModified()));
+		statusCode = HttpStatus.SC_OK;
+	}
+
+	private void getDirAsHttpResponse(File file) {
+		content = list2html(file);		
+		metadata.setValue(HttpHeaders.CONTENT_LENGTH, new Integer(content.length).toString());
+		metadata.setValue(HttpHeaders.CONTENT_TYPE, "text/html");
+		metadata.setValue(HttpHeaders.LAST_MODIFIED, formatDate(file.lastModified()));
+		statusCode = HttpStatus.SC_OK;
+	}
+
+	private String formatDate(long date) {
+
+		return dateFormat.format(new Date(date));
+	}
+
+	private byte[] list2html(File dir) {
+		
+		File[] files = dir.listFiles();
+		StringBuilder sb=new StringBuilder("<html><head>");
+		sb.append("<title>"+dir.getName()+"</title></head>\n");

```

Comment:
what's the point of having a StringBuilder if you don't use append?
`sb.append("<title>").append(dir.getName()).append("</title></head>\n")`;